lib/netproto/netproto_connect.c

```diff
@@ -105,11 +116,16 @@ callback_cancel(void * cookie)
 static struct sock_addr **
 getserveraddr(void)
 {
-	static struct sock_addr ** srv_addr = NULL;
 	static time_t srv_time = (time_t)(-1);
 	struct sock_addr ** tmp_addr;
 	time_t tmp_time;
 
+	/* This should only happen once. */
```

Comment:
I wasn't certain if there was any multi-threading issues here.  I mean, how long can `sock_resolve` take?  with the callbacks, I wasn't certain if you might call `getserveraddr` multiple times (before the first had finished).

However, I see that you don't use `pthread.h`, so I'll go ahead and make this change.  :)